FindmallocMC.cmake

```diff
@@ -158,5 +178,29 @@ find_package_handle_standard_args(mallocMC
     REQUIRED_VARS ${mallocMC_REQUIRED_VARS_LIST}
     VERSION_VAR mallocMC_VERSION
     HANDLE_COMPONENTS
-    )
+)
+
+# clean up
+#
+# unset cached variables in case we did not find a valid install
+# (e.g., we only found an outdated version)
+if(NOT mallocMC_FOUND)
+    # default vars
+    unset(mallocMC_VERSION)
+    foreach(REQ_VAR ${mallocMC_REQUIRED_VARS_LIST})
+        unset(${REQ_VAR})
+        unset(${REQ_VAR} CACHE)
+    endforeach()
+
+    # user-level component vars
+    foreach(COMPONENT ${mallocMC_FIND_COMPONENTS})
+        if(mallocMC_${COMPONENT}_FOUND)
```

Comment:
Yes, CMake did a wonderful job to ease the job of a developer that wants to use their component feature :poodle: